src/__tests__/Details.tsx

```diff
@@ -92,7 +93,7 @@ describe('Details', () => {
 
   it('Does not toggle when you click inside', () => {
     const Component = () => {
-      const {getDetailsProps} = useDetails({closeOnOutsideClick: true, defaultOpen: true})
+      const {getDetailsProps, open} = useDetails({closeOnOutsideClick: true, defaultOpen: true})
       return (
         <Details {...getDetailsProps}>
```

Comment:
Interesting. It looks like none of the Details tests are using the `getDetailsProps` function correctly. They all need to updated to call the function and pass the return value to the Details component:

```suggestion
        <Details {...getDetailsProps()}>
```

Reply:
You'll also probably need to update the prop type of Details to accept the return value of `getDetailsProps`

Reply:
Tried to accomplish this in https://github.com/primer/components/pull/1026/commits/57dafdda03b40868dc1976494713d506a864a096 and I'm getting a TS error. About to sign off here, but I might sync up with you in the morning to see if we can take a look at that.